libconcurrent/config.h

```diff
@@ -5,74 +5,73 @@
+// for more details.
 //#define DEBUG
 
 // Definition: DISABLE_BACKOFF
 // ---------------------------
-// By defining this, any backoff scheme used by any algorithm
-// is disabled. Be careful, upper an lower bounds must be
-// used in benchmark scripts, but they are ignored.
+// By defining this, any backoff scheme used by any algorithm is disabled. Be careful,
+// upper a lower bounds must be used in benchmark scripts, but they are ignored.
 //#define DISABLE_BACKOFF
 
 #define Object                     int64_t
 
 // Definition: RetVal
 // ------------------
-// Define the type of the return value that simulated 
-// atomic objects return. Be careful, it is assumed 
-// that the target architecture is able to atomically
-// read/write this type.
-// If this type is of 32 or 64 bits could be atomically 
-// read or written by most machine architectures.
-// However, a value of 128 bits or more may not be supported
-// (in most cases i.e. x86_64 supports types of 128 bits).
+// Define the type of the return value that simulated atomic objects return. Be careful,
+// it is assumed that the target architecture is able to atomically read/write this type.
+// If this type is of 32 or 64 bits could be atomically read or written by most machine
+// architectures. However, a value of 128 bits or more may not be supported(in most cases
+// x86_64 supports types of 128 bits).
 #define RetVal                     int64_t
 
 // Definition: ArgVal
 // ------------------
-// Define the type of the argument value of atomic objects.
-// All atomic objects have same argument types. In case
-// that you 'd like to use different argument values in each
+// Define the type of the argument value of atomic objects. All atomic objects have same
+// argument types. In case that you 'd like to use different argument values in each
 // atomic object, redefine it in object's source file.
 #define ArgVal                     int64_t
 
 #define NUMA_SUPPORT
 
 // Definition: SYNCH_COMPACT_ALLOCATION
 // ------------------------------------
-// This definition enables some experimental code on memory allocation that 
-// seems to greately improve the performance on AMD Epyc multiprocessors.
-// This flag seems to double the performance in CC-Synch and H-Synch algorithms.
+// This definition enables some optimizations on memory allocation that seems to greatly
+// improve the performance on AMD Epyc multiprocessors. In this case, this flag seems to
+// double the performance in CC-Synch and H-Synch algorithms. It is recommended to turn 
+// off this flag on Intel machines.
 #define SYNCH_COMPACT_ALLOCATION
 
+// Definition: POOL_NODE_RECYCLING_DISABLE
+// ---------------------------------------
+// This definition disables node recycling in the concurrent stack and queue
+// implementations that support memory reclamation. This may have negative impact in
+// performance. By default, this flag is enabled.
```

Comment:
Is there a rule that the user can follow to know whether to disable this or not?

Reply:
In terms of performance this pool implementation is a bit experimental.
Thus, the user should have this disabled whenever use data-structures for bench-marking.
For developing normal applications, this option should be considered safe.